services/workflows-service/src/transaction/transaction.controller.external.ts

```diff
@@ -275,4 +283,133 @@ export class TransactionControllerExternal {
       },
     });
   }
+
+  @Get()
+  @UseGuards(CustomerAuthGuard)
+  @swagger.ApiOkResponse({ description: 'Returns an array of transactions.' })
+  @swagger.ApiQuery({ name: 'businessId', description: 'Filter by business ID.', required: false })
+  @swagger.ApiQuery({
+    name: 'counterpartyId',
+    description: 'Filter by counterparty ID.',
+    required: false,
+  })
+  @swagger.ApiQuery({
+    name: 'startDate',
+    type: Date,
+    description: 'Filter by transactions after or on this date.',
+    required: false,
+  })
+  @swagger.ApiQuery({
+    name: 'endDate',
+    type: Date,
+    description: 'Filter by transactions before or on this date.',
+    required: false,
+  })
+  @swagger.ApiQuery({
+    name: 'paymentMethod',
+    description: 'Filter by payment method.',
+    required: false,
+    enum: PaymentMethod,
+  })
+  @swagger.ApiQuery({
+    name: 'timeValue',
+    type: 'number',
+    description: 'Number of time units to filter on',
+    required: false,
+  })
+  @swagger.ApiQuery({
+    name: 'timeUnit',
+    type: 'enum',
+    enum: Object.values(TIME_UNITS),
+    description: 'The time unit used in conjunction with timeValue',
+    required: false,
+  })
+  @swagger.ApiQuery({
+    name: 'alertId',
+    description: 'Filter by alert ID.',
+    required: true,
+  })
+  async getTransactionsByAlert(
+    @Query() getTransactionsByAlertParameters: GetTransactionsByAlertDto,
+    @CurrentProject() projectId: types.TProjectId,
+  ) {
+    const alert = await this.prisma.alert.findUnique({
+      where: {
+        id: getTransactionsByAlertParameters.alertId,
+      },
+    });
```

Comment:
you can use include workflow def and you improve performance by querying the database only once

Reply:
fixed